internal/renderer/service_util.go

```diff
@@ -341,6 +343,51 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 		})
 	}
 
+	// Set health check port
+	regexProtocol := `^service\.beta\.kubernetes\.io\/.*health.*protocol$`
+	regexPort := `^service\.beta\.kubernetes\.io\/.*health.*port$`
+
+	annotationRegexProtocol := regexp.MustCompile(regexProtocol)
+	annotationRegexPort := regexp.MustCompile(regexPort)
+
+	var healthCheckPort int32
+	healthCheckProtocol := "TCP"
+
+	// Find health check port
+	for annotationKey, annotation := range c.gwConf.Spec.LoadBalancerServiceAnnotations {
```

Comment:
Bad news: we can have annotations both in the GwConf (as you're handing above) as well as on individual Gateways (which we do not handle here). I know it's stupid: I'd like to deprecate `GwConf.LoadBalancerServiceAnnotations` ASAP but this will not happen before we release v1.0. Until then, we need to handle both.

What I'd recommend is to copy the GwConfig.LBSvcAnnotations into a new map, then merge the GatewayAnnotations into this, and then do the regexp matches on the merged map. This correctly reproduces the intended semantics, that global GwConfig.LBSvcAnnotations are overridden by per-Gw annotations.